contributors/design-proposals/vertical-pod-autoscaler.md

```diff
@@ -0,0 +1,660 @@
+	// +optional
+	UpdatePolicy PodUpdatePolicy
+
+	// ResourcePolicy controls how the autoscaler computes recommended resources.
+	// +optional
+	ResourcePolicy PodResourcePolicy
+
+	// Recommendation is the most recently computed amount of resources
+	// recommended by the autoscaler for the controlled pods.
+	// +optional
+	Recommendation RecommendedPodResources
+}
+
+// VerticalPodAutoscalerTarget determines the set of pods controlled by the
+// vertical pod autoscaler.
+type VerticalPodAutoscalerTarget struct {
+	// PodSelector is the label selector for pods to match.
+	// (see apimachinery/pkg/labels/selector.go).
+	PodSelector string
+}
+
+// UpdateMode controls when autoscaler applies changes to the pod resoures.
+type UpdateMode string
+const (
+	// UpdateModeOff means that autoscaler never changes Pod resources.
+	// The recommender still sets the recommended resources in the
+	// VerticalPodAutoscaler object. This can be used for a "dry run".
+	UpdateModeOff UpdateMode = "Off"
+	// UpdateModeInitial means that autoscaler only assigns resources on pod
+	// creation and does not change them during the lifetime of the pod.
+	UpdateModeInitial UpdateMode = "Initial"
+	// UpdateModeAuto means that autoscaler assigns resources on pod creation
+	// and additionally can update them during the lifetime of the pod,
+	// including evicting / rescheduling the pod.
+	UpdateModeAuto UpdateMode = "Auto"
+)
+
+// PodUpdatePolicy describes the rules on how changes are applied to the pods.
+type PodUpdatePolicy struct {
+	// UpdateMode controls when autoscaler applies changes to the pod resoures.
+	// +optional
+	UpdateMode UpdateMode
+}
+
+// ContainerResourcePolicy controls how autoscaler computes the recommended
+// resources for a specific container.
+type ContainerResourcePolicy struct {
+	// MinAllowed specifies the minimal amount of resources that will be
+	// recommended for the container.
+	// +optional
+	MinAllowed api.ResourceRequirements
+	// MaxAllowed specifies the maximum amount of resources that will be
+	// recommended for the container.
+	// +optional
+	MaxAllowed api.ResourceRequirements
+}
+
+// PodResourcePolicy controls how autoscaler computes the recommended resources
+// for containers belonging to the pod.
+type PodResourcePolicy struct {
+	Containers []ContainerResourcePolicy
+}
+
+// RecommendedPodResources is the recommendation of resources computed by
+// autoscaler.
+type RecommendedPodResources struct {
+	// ContainerRecommendations are resources recommended by autoscaler for each
+	// container.
+	ContainerRecommendations []RecommendedContainerResources
+}
+
+// RecommendedContainerResources is the recommendation of resources computed by
+// autoscaler for a specific container.
+type RecommendedContainerResources struct {
+	// Name is the name of the container.
+	Name string
+	// MinRecommended is the recommended minimum amount of resources.
+	MinRecommended api.ResourceRequirements
```

Comment:
There should probably be the direct target oneapi.ResourceRequirements, so that initializers/updaters don't need to think too much. Min and Max are for proper bounding/more forgiving updates.